### ISSUE TITLE
Explain ordering of prompt embeds

### ISSUE THREAD

**Issue by jmitrovicTT** [tenstorrent/tt-metal#26620]
## Description

From when we started using data parallelism, the accessing of prompt_embeds and negative_prompt_embeds has changed across commits.

Essentially, before commit 5f2d035 (SDXL trace implementation #24300) reordering of prompt embeddings was required, while after that commit reordering is no longer necessary.

I will illustrate this with **prompt_embeds** and **negative_prompt_embeds** — for pooled_prompt_embeds and negative_pooled_prompt_embeds the situation is equivalent, only the tensor dimensions differ.

### Before SDXL trace implementation

We start with a list of tensors for `prompt_embeds` and `negative_prompt_embeds`, `N` is the `num_prompts + num_padding_prompts` so it is always divisible by `batch_size`:  

 - initial dimensions: `[N * [1, 77, 2048]]`   

Concatenation is performed: ⬇️  `torch.cat(prompt_embeds, dim=0)`    

 - resulting tensor after `torch.cat`:`[N, 77, 2048]`  

Stack function is performed: ⬇️ `stacked_embeds` = `torch.stack([negative_prompt_embeds, prompt_embeds], dim=1)`

 - tensor after `torch.stack`: [N, 2, 77, 2048], - because `torch.stack` is done with `dim=1` so: 
    - `stacked_embeds[:, 0, :, "]` == `negative_prompt_embeds`
    - `stacked_embeds[:, 1, :, "]` == `prompt_embeds`

stacked tensor is sharded on DRAM ⬇️ with dim=0: 
```
    ttnn_prompt_embeds = ttnn.from_torch(
        torch_prompt_embeds,
        dtype=ttnn.bfloat16,
        device=ttnn_device,
        layout=ttnn.TILE_LAYOUT,
        memory_config=ttnn.DRAM_MEMORY_CONFIG,
        mesh_mapper=ttnn.ShardTensorToMesh(ttnn_device, dim=0),
    )
```

 - resulting sharded tensor has dimension: `[N//batch_size, 2, 77, 2048]`,
 
problem was that sharding tensor to mesh don't follow ordering logic: **example with 32 cores => `batch_size = 32` and `N = 1024`**:

| core 1 | core 2 | ... | core 32 |
|------|--------------|-----------|-------|
| embed_0    | embed_32      | ...      | embed_992 |
| embed_1    | embed_33      | ...      | embed_993 |
| ...    | ...      | ...      | ... |
| embed_31    | embed_63      | ...      | embed_1023 |
---

in `run_tt_image_gen` function we called ⬇️ `run_tt_iteration` for each batch of embeds this way:  ⬇️
- `tt_prompt_embeds[iter][unet_slice]`
where **`iter` meant index of batch from `0 to N // batch_size`** and **`unet_slice` meant index accessing negative prompt or positive prompt_embeds.**

so for `iter=0` for example we generated images for embeds:
`[embed_0,  embed_32,  ... , embed_992 ]` - so ** CLIP score was bad and we needed reordering **

### After SDXL trace implementation

Here we start the same - we have a list of tensors and after `torch.cat` we have result tensor with dimensions: `[N, 77, 2048]` both for `prompt_embeds` and `negative_prompt_embeds`.

now `torch.split` is performed on `dim=0` and `batch_size`⬇️:
```
prompt_embeds = torch.split(prompt_embeds_cat, batch_size, dim=0)
negative_prompt_embeds = torch.split(negative_prompt_embeds_cat, batch_size, dim=0)
```

and resulting structures are tuples of `N//batch_size` tensors each with dimension `[batch_size, 77, 2048]` for both `prompt_embeds` and `negative_prompt_embeds`.

in `create_user_tensors` function  ⬇️

```
# in **create_user_tensors** function
tt_prompt_embeds = [
    [
        ttnn.from_torch(
            negative_prompt_embed,
            dtype=ttnn.bfloat16,
            layout=ttnn.TILE_LAYOUT,
            mesh_mapper=ttnn.ShardTensorToMesh(ttnn_device, dim=0) if is_mesh_device else None,
        ),
        ttnn.from_torch(
            prompt_embed,
            dtype=ttnn.bfloat16,
            layout=ttnn.TILE_LAYOUT,
            mesh_mapper=ttnn.ShardTensorToMesh(ttnn_device, dim=0) if is_mesh_device else None,
        ),
    ]
    for negative_prompt_embed, prompt_embed in zip(negative_prompt_embeds, prompt_embeds)
]
```

structure created is list of `N//batch_size` lists of `2` tensors:
`[N//batch_size*[2*[batch_size, 77, 2048]]]`, and tensors are prepared for sharding.

 - Note: Trace-enabled execution requires that all tensors used during tracing remain at **fixed memory locations** so that the trace can later replay them correctly.  
This is why we need to allocate the input tensor once via `allocate_input_tensor`: it provides a reserved memory region that will be reused for every batch of prompts.  
Before each batch, the current `batch of negative and positive prompt embeds`is copied into this fixed location with `prepare_input_tensors` function.

in `allocate_input_tensors` function: in DRAM is alocated:
`[2*[1, 77, 2048]]` - which represents:
`[[1, 77, 2048] - sharded_negative_embed, [1, 77, 2048]-sharded_positive_embed]` ⬇️

```
# in **allocate_input_tensors** function
tt_prompt_embeds_device = [
        ttnn.allocate_tensor_on_device(
            tt_prompt_embeds[0][0].shape,
            tt_prompt_embeds[0][0].dtype,
            tt_prompt_embeds[0][0].layout,
            ttnn_device,
            ttnn.DRAM_MEMORY_CONFIG,
        ),
        ttnn.allocate_tensor_on_device(
            tt_prompt_embeds[0][1].shape,
            tt_prompt_embeds[0][1].dtype,
            tt_prompt_embeds[0][1].layout,
            ttnn_device,
            ttnn.DRAM_MEMORY_CONFIG,
        ),
    ]
```

so before each `run_tt_image_gen` function, `prepare_input_tensors` is called on `tt_prompt_embeds[iter]` which copies current batch of embeds in that alocated memory (also sharding is applied in this step) - this mechanism is needed tor tracing - and now used even if `test_demo` is called with `"no_trace"` parameter.

now each `run_tt_iteration` works with exactly one sharded batch of prompt_embeds:
`tt_prompt_embeds[unet_slice]` where with unet_slice we access sharded negative_prompt_embeds or sharded positive_prompt_embeds.

So now with this setup reordering is not needed anymore.

Tasks:
- [ ] Explain ordering of prompt embeds
- [ ] Delete reordering logic
- [ ] Add check that fails accuracy_test if CLIP scores are out of reference range
- [ ] Include MLPerf hyperparameters in our test runs
- [ ] Add JSON unique ID
